pkg/service/v1alpha1/controller.go

```diff
@@ -99,6 +102,10 @@ func (cs *controller) CreateVolume(
 	VolumeContext := map[string]string{
 		"openebs.io/cas-type": req.GetParameters()["cas-type"],
 	}
+	if req.GetAccessibilityRequirements() != nil {
+		nodeID = req.GetAccessibilityRequirements().
+			GetPreferred()[0].GetSegments()[HostTopologyKey]
```

Comment:
i think we have to give random node name instead always pick the first one, in generic provisioning ( without WaitForFirstConsumer)

Reply:
We have to document this as well.